implementations/rust/ockam/ockam_abac/src/storage/resource_policy_repository.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{Action, Expr, ResourceName, ResourcePolicy, ResourceType};
+use ockam_core::async_trait;
+use ockam_core::compat::boxed::Box;
+use ockam_core::compat::vec::Vec;
+use ockam_core::Result;
+
+/// This repository stores policies for resources.
+/// A policy is an expression which can be evaluated against an environment (a list of attribute
+/// names and values) in order to determine if a given action can be performed on a given resource.
+#[async_trait]
+pub trait ResourcePoliciesRepository: Send + Sync + 'static {
+    /// Set a policy for a given resource and action
+    async fn set_policy(
+        &self,
+        resource_name: &ResourceName,
+        resource_type: Option<&ResourceType>,
```

Comment:
Could you please explain why we need resource_type in this repository?

Reply:
We know the resource_name + resource_type pair only when creating an outlet or an inlet. Once we are done with that operation, how do we know what global policy should apply to the policy attached to an outlet? Before this change, we only had this information inside the `create_inlet|outlet` functions.

Having this extra column allows us to figure out the final expression for a resource policy from anywhere in the codebase. This is especially helpful from the command line: without it, the user would have to specify both the resource name and the resource type (and I don't know how they would figure this out) to retrieve the correct policy expression for that resource.